### DISCUSSION TITLE
pnpm install econnrefused behind proxy against azure devops registry

### DISCUSSION THREAD

**Discussion by Link631** [pnpm/pnpm#4336] [Q&A]
I want to swtich from npm to pnpm.

with npm everything works fine, I can connect and install packages from my private registry hosted in azure devops with credentials being conciguered through vsts-npm-auth.

When i switch from npm to pnpm i can see, that pnpm tries to install packages from azure devops, but it gets an econnrefused.

Why?

Shouldnt it be working, when npm is working fine?

**Comment by zkochan** [comment]
Yes, pnpm should read the proxy configuration from the npm configuration files.

**Comment by Link631** [comment]
ok...

and what about the auth information, which npm stores into .npmrc file with vsts-npm-auth, ie. something like this:

proxy=xyz
https-proxy=xyz
HTTP_PROXY=xyz
HTTPS_PROXY=xyz
noproxy[]=localhost
//pkgs.dev.azure.com/..../_packaging/Node/npm/registry/:username=VssSessionToken
; This is an unencrypted authentication token. Treat it like a password. DO NOT share this value with anyone, including Microsoft support.
//pkgs.dev.azure.com/.../_packaging/Node/npm/registry/:_password=generatedPassword==
; The npm client won't use username and _password without an email set, but the service doesn't use the email value. The following is an arbitrarily made-up address.
//pkgs.dev.azure.com/..../_packaging/Node/npm/registry/:email=not-used@example.com

**Reply by zkochan** [reply]
yes, this should also work.

**Comment by Link631** [comment]
hmm....

any Idea why this still happens with the mentioned configuration plus this config:

....
registry=https://pkgs.dev.azure.com/.../_packaging/Node/npm/registry/
always-auth=true

Result:

pnpm install react

GET https://pkgs.dev.azure.com/.../_packaging/Node/npm/registry/react error (ECONNREFUSED). Will retry in 10 seconds. 2 retries left.

npm install works just fine... :-(
